### PR TITLE
fix(52664): Assertion failure on completions for derived class with computed base property name

### LINKED ISSUE #52664: Assertion failure on completions for derived class with computed base property name
```ts
export const SOME_CONSTANT = 'SOME_TEXT';

export class Base {
    [SOME_CONSTANT]: boolean;
}

export class Derived extends Base {
    /*$*/
}
```

Request class snippet completions at `/*$*/`.

```
Message: Error processing request. Debug Failure.
Error: Debug Failure.
    at Object.getTextOfPropertyName (/vscode/bin/linux-x64/some-id-insider/extensions/node_modules/typescript/lib/tsserver.js:15429:25)
    at Object.getMemberOverrideModifierStatus (/vscode/bin/linux-x64/some-id-insider/extensions/node_modules/typescript/lib/tsserver.js:86231:63)
    at /vscode/bin/linux-x64/some-id-insider/extensions/node_modules/typescript/lib/tsserver.js:135765:32
    at Object.addNewNodeForMemberSymbol (/vscode/bin/linux-x64/some-id-insider/extensions/node_modules/typescript/lib/tsserver.js:161425:21)
    at getEntryForMemberCompletion (/vscode/bin/linux-x64/some-id-insider/extensions/node_modules/typescript/lib/tsserver.js:135751:24)
    at createCompletionEntry (/vscode/bin/linux-x64/some-id-insider/extensions/node_modules/typescript/lib/tsserver.js:135603:23)
    at getCompletionEntriesFromSymbols (/vscode/bin/linux-x64/some-id-insider/extensions/node_modules/typescript/lib/tsserver.js:136106:29)
    at completionInfoFromData (/vscode/bin/linux-x64/some-id-insider/extensions/node_modules/typescript/lib/tsserver.js:135415:31)
    at Object.getCompletionsAtPosition (/vscode/bin/linux-x64/some-id-insider/extensions/node_modules/typescript/lib/tsserver.js:135275:36)
    at Object.getCompletionsAtPosition (/vscode/bin/linux-x64/some-id-insider/extensions/node_modules/typescript/lib/tsserver.js:169151:35)
    at IpcIOSession.Session.getCompletions (/vscode/bin/linux-x64/some-id-insider/extensions/node_modules/typescript/lib/tsserver.js:181300:64)
    at Session.handlers.ts.Map.ts.getEntries._a.<computed> (/vscode/bin/linux-x64/some-id-insider/extensions/node_modules/typescript/lib/tsserver.js:179973:61)
    at /vscode/bin/linux-x64/some-id-insider/extensions/node_modules/typescript/lib/tsserver.js:182034:96
    at IpcIOSession.Session.executeWithRequestId (/vscode/bin/linux-x64/some-id-insider/extensions/node_modules/typescript/lib/tsserver.js:182025:28)
    at IpcIOSession.Session.executeCommand (/vscode/bin/linux-x64/some-id-insider/extensions/node_modules/typescript/lib/tsserver.js:182034:41)
    at IpcIOSession.Session.onMessage (/vscode/bin/linux-x64/some-id-insider/extensions/node_modules/typescript/lib/tsserver.js:182062:35)
    at process.<anonymous> (/vscode/bin/linux-x64/some-id-insider/extensions/node_modules/typescript/lib/tsserver.js:186218:31)
    at process.emit (node:events:526:28)
    at emit (node:internal/child_process:938:14)
    at processTicksAndRejections (node:internal/process/task_queues:84:21)
```

### PR DESCRIPTION
**Pull request by a-tarasyuk**
Fixes #52664

### CODE REVIEW

File: tests/baselines/reference/completionsClassMembers4.baseline [hunk 1]
```diff
@@ -0,0 +1,254 @@
...
+          "sortText": "15",
+          "displayParts": [
+            {
+              "text": "static",
+              "kind": "keyword"
+            }
+          ]
+        },
+        {
+          "name": "SOME_TEXT",
```
Comment (a-tarasyuk) [inline tests/baselines/reference/completionsClassMembers4.baseline:197]: @DanielRosenwasser Do we need to add brackets for computed (`["SOME_TEXT"]`, `[SOME_CONSTANT]`?) inherited members? [Playground](https://www.typescriptlang.org/play?#code/KYDwDg9gTgLgBAYwgOwM7wMoHkCyBRAfQGEsA5DAFQEFSK4BeOAcm3wIrwA0KmBuAKFCRYiADYBDVKjgAhScDgBvfnFVwA2q0Ily1WgF0AXHABGECKODjkDOADNxo1MAEBfQeGjwEEqXAAiwFAAlgBuwAAmcKAwwMgR0nLOSipq-O5AA)

Comment (DanielRosenwasser) [reply tests/baselines/reference/completionsClassMembers4.baseline:197]: Probably better to write `[SOME_CONSTANT]` if it's reasonable. `["SOME_TEXT"]` is a good alternative.

**Review by DanielRosenwasser — APPROVED** [review]

File: src/compiler/checker.ts [hunk 2]
```diff
@@ -42624,17 +42624,18 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
      * Checks a member declaration node to see if has a missing or invalid `override` modifier.
      * @param node Class-like node where the member is declared.
      * @param member Member declaration node.
+     * @param symbol Member symbol.
      * Note: `member` can be a synthetic node without a parent.
      */
-    function getMemberOverrideModifierStatus(node: ClassLikeDeclaration, member: ClassElement): MemberOverrideStatus {
+    function getMemberOverrideModifierStatus(node: ClassLikeDeclaration, member: ClassElement, symbol: Symbol): MemberOverrideStatus {
```
Comment (DanielRosenwasser) [inline src/compiler/checker.ts:42630]: maybe `symbol` -> `memberSymbol`? and is it the base member symbol?

**Comment by gabritto** [comment]
I'm not sure if this PR is still in progress, but @a-tarasyuk could you add some tests? There are a few things in this PR I couldn't fully understand, and I think tests could help me review it. Thanks.

**Comment by DanielRosenwasser** [comment]
I think the lack of tests comes from a merge conflict or something. There was a test previously at https://github.com/microsoft/TypeScript/pull/52673/commits/d0c5db8cebbe3bd36fe6712e75ca0076b8a63658#diff-8bc13ef2c947f0e85e6a6d61129663eb774aa9f40a694b2cf8f85a4757a75c4b

**Comment by gabritto** [comment]
> I think the lack of tests comes from a merge conflict or something. There was a test previously at [d0c5db8#diff-8bc13ef2c947f0e85e6a6d61129663eb774aa9f40a694b2cf8f85a4757a75c4b](https://github.com/microsoft/TypeScript/commit/d0c5db8cebbe3bd36fe6712e75ca0076b8a63658#diff-8bc13ef2c947f0e85e6a6d61129663eb774aa9f40a694b2cf8f85a4757a75c4b)

Ah, that explains it. @a-tarasyuk can you re-add the test then?

**Review by jakebailey — APPROVED** [review]

**Pull request merged**